### PR TITLE
catalog-node: Export experimental catalogServiceRef

### PR DESCRIPTION
**Pull request by jhaals**
Adds experimental catalogServiceRef for obtaining a `catalogClient` in the new backend system

### CODE REVIEW

**Comment by jhaals** [comment]
I think we should do something to simplify testing of service factories.

File: .changeset/chatty-cars-kick.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-catalog-node': patch
+---
+
+Adds experimental catalogServiceRef for obtaining a `catalogClient` in the new backend system
```
Comment (Rugvip) [inline .changeset/chatty-cars-kick.md:5]:
```suggestion
Adds experimental `catalogServiceRef` for obtaining a `CatalogClient` in the new backend system.
```

File: plugins/scaffolder-backend/src/ScaffolderPlugin.ts [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+import { TemplateFilter } from './lib';
+import { createBuiltinActions, TaskBroker, TemplateAction } from './scaffolder';
+import { createRouter } from './service/router';
+
+/**
+ * Catalog plugin options
+ * @alpha
+ */
+export type ScaffolderPluginOptions = {
+  actions?: TemplateAction<any>[];
```
Comment (freben) [inline plugins/scaffolder-backend/src/ScaffolderPlugin.ts:38]: these <any> should proably have been <unknown> in my mind

**Review by freben — APPROVED** [review]

**Review by Rugvip — APPROVED** [review]

File: plugins/scaffolder-backend/src/ScaffolderPlugin.ts [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    this.#actions.push(...actions);
+  }
+  get actions() {
+    return this.#actions;
+  }
+}
+
+/**
+ * @alpha
+ * TODO: MOVE to catalog-node before merge.
```
Comment (Rugvip) [inline plugins/scaffolder-backend/src/ScaffolderPlugin.ts:66]:
```suggestion
 * TODO: MOVE to scaffolder-node.
 */
interface ScaffolderActionsExtensionPoint {
  addActions(...actions: TemplateAction<any>[]): void;
}

class ScaffolderActionsExtensionPointImpl
  implements ScaffolderActionsExtensionPoint
{
  #actions = new Array<TemplateAction<any>>();
  addActions(...actions: TemplateAction<any>[]): void {
    this.#actions.push(...actions);
  }
  get actions() {
    return this.#actions;
  }
}

/**
 * @alpha
 * TODO: MOVE to scaffolder-node.
```

Comment (freben) [reply plugins/scaffolder-backend/src/ScaffolderPlugin.ts]: dat diff

**Pull request merged**